pkg/operator/ceph/object/spec.go

```diff
@@ -406,14 +406,24 @@ func (c *clusterConfig) vaultPrefixRGW() string {
 	switch secretEngine {
 	case kms.VaultKVSecretEngineKey:
 		vaultPrefixPath = path.Join(vaultPrefixPath,
-			c.clusterSpec.Security.KeyManagementService.ConnectionDetails[vault.VaultBackendPathKey])
+			c.clusterSpec.Security.KeyManagementService.ConnectionDetails[vault.VaultBackendPathKey], "/data")
 	case kms.VaultTransitSecretEngineKey:
 		vaultPrefixPath = path.Join(vaultPrefixPath, secretEngine, "/export/encryption-key")
 	}
 
 	return vaultPrefixPath
 }
 
+func (c *clusterConfig) sseIsEnabledRGW() bool {
+	secretEngine := c.clusterSpec.Security.KeyManagementService.ConnectionDetails[kms.VaultSecretEngineKey]
+	kvvers := c.clusterSpec.Security.KeyManagementService.ConnectionDetails[vault.VaultBackendKey]
+	if secretEngine == kms.VaultKVSecretEngineKey && kvvers != "v2" {
+		logger.Warning("kv version v1 is not supported for RGW, hence disabling server side encryption")
```

Comment:
Should this be an error instead? Users have requested encryption in the config; we probably do not want to disable encryption with nothing more than a warning which they can easily miss. IMO, we should throw an error here and report that they should either choose "v2" protocol or disable encryption.

Reply:
@BlaineEXE : So in this case RGW pod shouldn't come up than disabling encryption